src/modules/SX126x/STM32WLx.h

```diff
@@ -0,0 +1,171 @@
+/*
+
+Copyright (c) 2018 Jan Gromeš
+Copyright (c) 2022 STMicroelectronics
+
+This file is licensed under the MIT License: https://opensource.org/licenses/MIT
+*/
+
+#if !defined(_RADIOLIB_STM32WLx_H)
+#define _RADIOLIB_STM32WLx_H
+
+#include "../../TypeDef.h"
+
+#if !defined(RADIOLIB_EXCLUDE_STM32WLX)
+
+#include "../../Module.h"
+#include "SX1262.h"
+#include "STM32WLx_Module.h"
+
+/*!
+  \class STM32WLx
+
+  \brief Derived class for STM32WL modules.
+
+  The radio integrated into these modules is essentially the same as the
+  Semtech %SX126x external radio chips, so most of the documentation for
+  those also applies here.
+
+  #### RF Switch and low-power vs high-power transmissions
```

Comment:
The reason I put this here, is that it deviates from the normal RfSwitch documentation, which uses `Module::MODE_*` constants. On STM32WL, separate mode constants and an extra mode is available, so this has be to documented somewhere. Putting this in the `STM32WLx::setRfSwitchTable()` docs would make sense, except that method does not exist (it just inherits the SX126x version without chagnes), so putting it in the class header seemed to most obvious place (though I guess we could explicitly inherit it with a `using` directive, or a method that just calls its parent, and document that perhaps).

Moving this into the wiki does not seem right to me, since then the API documentation in the code is missing relevant details about how to use the rf switch and how to set up modules that only support either LP or HP. Maybe duplicating (instead of moving) some of that information in the wiki could be a good idea, though I don't see an obvious place for this directly, https://github.com/jgromes/RadioLib/wiki/Modules only has very concise info for different module types).

Reply:
I think this is a bit more high-level description of how the STM32WL functions, not so much related to any single method in STM32WLx.

The API documentation is also missing information about other high-level things, such as "how do you send a message?" - that's what the examples are for. And for even more generic documentation (such as https://github.com/jgromes/RadioLib/wiki/High-power-Radio-Modules-Guide), there's the wiki. In general I would say that the purpose of the doxygen docs is to describe "how do I use this method", as opposed to "how do I use this device".

Reply:
> The API documentation is also missing information about other high-level things, such as "how do you send a message?"

This is true, but in this case a signficant part of the documentation is about how the API for *this* module differs from the generic / SX126x module, which is otherwise lost in the API docs.

What I did now was add documentation for the `STM32WLx::setRfSwitchTable()` method, focusing on the API and not the module workings. The documentation in the `STM32WLx` class header was removed, leaving just a small remark about the amplifier difference and which APIs this affects. I preserved the original comment in the first post of this PR for later use on e.g. the wiki if needed.

See 5cc28361ceed9c1c569dcb16bc34f584d0e77966.